src/charm.py

```diff
@@ -31,8 +36,31 @@ def _on_install(self, event: InstallEvent) -> None:
         Returns:
             None
         """
+        self.unit.status = MaintenanceStatus("Installing AGW Snap")
         self.install_magma_access_gateway_snap()
+        if not self._is_configuration_valid():
+            self.unit.status = BlockedStatus("Configuration is invalid. Check logs for details")
+            event.defer()
+            return
+        self.unit.status = MaintenanceStatus("Installing AGW")
         self.install_magma_access_gateway()
+
+    def _on_start(self, event: StartEvent):
+        """Truggered on start event.
+
+        Args:
+            event: Juju event
+
+        Returns:
+            None
+        """
+        magma_service = subprocess.run(
+            ["systemctl", "is-active", "magma@magmad"],
+            stdout=subprocess.PIPE,
+        )
+        if magma_service.returncode != 0:
+            event.defer()
```

Comment:
Is there a reason why the second/third/... time it would work vs. the first?

Reply:
Yes, the reason is that the install script that we call from the snap does potentially 2 reboots. It can do a reboot first to reconfigure the network, and it does a reboot after installing all the services.

During all that time, the status will stay to `MaintenanceStatus("Installing AGW")`. A reboot with the magmad service active is the indicator that the process has completed.